src/services/autoMailer.js

```diff
@@ -0,0 +1,31 @@
+import sgMail from '@sendgrid/mail';
+import dotenv from 'dotenv';
+import emailTemplatesFunction from '../utils';
+
+dotenv.config();
+
+sgMail.setApiKey(process.env.SENDGRID_API_KEY);
+
+const sendEmail = async (receiver, category, data) => {
+  try {
+    const retrievedData = emailTemplatesFunction(category, data);
+    const {
+      html,
+      subject
+    } = retrievedData;
+    const msg = {
+      to: receiver,
+      from: 'wolfsbane@barefootnomad.herokuapp.com',
+      subject,
+      html,
+    };
+    const result = await sgMail.send(msg);
+    if (result[0] && result[0].request) {
+      return true;
+    }
+  } catch (error) {
```

Comment:
Also, like @uthdev said, the tests for sending the email are failing. So I did a few console.log statements to trace the error source, and I found out that the `sendEmail` function returns a `401 unauthorized` error here. I tried changing the `from` email, maybe it was because that wasn't the email you registered with, but I got the same error. My other guess is probably we got the wrong API KEY. Can you help us look into this and get back? Thanks.

Reply:
This conversation was marked as resolved, please I will need the information of how it was resolved